src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs

```diff
@@ -2216,8 +2240,16 @@ private void ClassifyInsert(SyntaxNode node)
                 switch (node.Kind())
                 {
                     case SyntaxKind.GlobalStatement:
-                        // TODO:
-                        ReportError(RudeEditKind.Insert);
+                        // Global statements can be inserted but only if there is at least one other global statement,
+                        // otherwise this would be an insert of the synthesized entry point itself.
+                        if ((node.Parent as CompilationUnitSyntax)?.Members.Count == 1)
+                        {
+                            ReportError(RudeEditKind.Insert);
+                            return;
+                        }
```

Comment:
Would that work with:

```
// var x = ""; // uncomment.

public class C { }
```

I think the count will be 2 after uncommenting, but it should be reported as rude edit.